libs/core/langchain_core/schema/document_loader.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Iterator, List, Optional
+
+from langchain_core.documents import Document
+from langchain_core.schema.text_splitter import TextSplitterInterface
+
+
+class DocumentLoaderInterface(ABC):
+    """Interface for Document Loader.
+
+    Implementations should implement the lazy-loading method using generators
+    to avoid loading all Documents into memory at once.
+
+    The `load` method will remain as is for backwards compatibility, but its
+    implementation should be just `list(self.lazy_load())`.
+    """
+
+    @abstractmethod
+    def load(self) -> List[Document]:
```

Comment:
Implementation should be provided by default and proxy lazy_load. Users should never be implementing `load` and ideally not using it either -- lots of enterprise use cases involve millions of docs, so these loaders will OOM

Reply:
If it is mandatory we can just remove `lazy_load` and leave `load` but enforce the lazy behavior with the lazy signature. Unfortunately, it looks like a braking change.